base/pkg/entry.jl

```diff
@@ -250,11 +250,11 @@ function free(pkg::AbstractString)
             for ver in vers
                 sha1 = avail[ver].sha1
                 LibGit2.iscommit(sha1, repo) || continue
-                return LibGit2.transact(repo) do r
+                LibGit2.transact(repo) do r
                     LibGit2.isdirty(repo) && throw(PkgError("$pkg is dirty, bailing"))
                     LibGit2.checkout!(repo, sha1)
-                    resolve()
                 end
+                return resolve()
```

Comment:
this version isn't right either, as if the checkout fails you should restore the previous state, and not run resolve. likewise if resolve fails after checkout, should roll back.

Reply:
I should have added a disclaimer that I blindly tried to fix the bug without being familiar with that code at all, so I wouldn't be surprised if that's totally incorrect. I can just note that the tests didn't cover that apparently, and that AFAICT the existing `free(pkgs)` method seems to work the same (but maybe I'm missing something). Anyway this commit doesn't need to be kept as removing the whole method also works.